src/main/java/com/ibm/crypto/plus/provider/XDHPrivateKeyImpl.java

```diff
@@ -55,9 +56,14 @@ final class XDHPrivateKeyImpl extends PKCS8Key implements XECPrivateKey, Seriali
     private transient XECKey xecKey = null;
 
     private void setFieldsFromXeckey() throws Exception {
-        if (this.key == null) {
-            this.key = extractPrivateKeyFromOCK(xecKey.getPrivateKeyBytes()); // Extract key from GSKit and sets params
-            this.scalar = Optional.of(key);
+        if (k == null) {
+            k = extractPrivateKeyFromOCK(xecKey.getPrivateKeyBytes()); // Extract key from GSKit and sets params
+            if (Integer.parseInt(provider.getVersionStr()) <= 11) {
+                this.key = k;
+            } else {
+                this.key = new DerValue(DerValue.tag_OctetString, k).toByteArray();
+            }
```

Comment:
Can this logic be broken out into a helper function? This same code seems to be used in 3 locations within this change.

Reply:
Done.